lib/carbon/http.py

```diff
@@ -1,74 +1,21 @@
-from twisted.internet import defer, protocol, reactor
-from twisted.web import client
-from twisted.web.http_headers import Headers
-from twisted.web.iweb import IBodyProducer
-from zope.interface import implements
-
-try:
-  from urllib import urlencode
-except ImportError:
-  from urllib.parse import urlencode
-
-
-class QuietHTTP11ClientFactory(client._HTTP11ClientFactory):
-    noisy = False
-
+import urllib3
 
 # async http client connection pool
-pool = client.HTTPConnectionPool(reactor)
-pool._factory = QuietHTTP11ClientFactory
-
-
-class StringProducer(object):
-  implements(IBodyProducer)
-
-  def __init__(self, body):
-    self.body = body
-    self.length = len(body)
-
-  def startProducing(self, consumer):
-    consumer.write(self.body)
-    return defer.succeed(None)
-
-  def pauseProducing(self):
-    pass
-
-  def stopProducing(self):
-    pass
-
-
-class SimpleReceiver(protocol.Protocol):
-  def __init__(self, response, d):
-    self.response = response
-    self.buf = ''
-    self.d = d
-
-  def dataReceived(self, data):
-    self.buf += data
-
-  def connectionLost(self, reason):
-    # TODO: test if reason is twisted.web.client.ResponseDone, if not, do an errback
-    self.d.callback({
-      'code': self.response.code,
-      'body': self.buf,
-    })
+http = urllib3.PoolManager()
 
 
-def httpRequest(url, values=None, headers=None, method='POST'):
-  fullHeaders = {
-    'Content-Type': ['application/x-www-form-urlencoded']
-  }
-  if headers:
-    fullHeaders.update(headers)
+def httpRequest(url, values=None, headers=None, method='POST', timeout=5):
```

Comment:
i'm wondering how it will affect performance, especially when GRAPHITE_URL is not configured or configured wrong. Will it block whole processing for 5 seconds?

Reply:
It should just block the tag writer thread, I'll verify that.

Reply:
Verified that it just blocks the tag writer thread

Reply:
👍